documentation/ContextManager.md

```diff
@@ -1,10 +1,110 @@
+
+The conditions supported by the system are: 
+* `http://registry.gpii.net/conditions/inRange`
+* `http://registry.gpii.net/conditions/timeInRange`
+
+These conditions along with their expected inputs are described below:
+
+#### http://registry.gpii.net/conditions/inRange
+
+This condition is declared using the `http://registry.gpii.net/conditions/inRange` type. It can be used for any condition which deals with a range in which a context value has to fall between. Besides the context, it can take two parameters; `min` and `max`. These expresses the values (inclusive) which the `inputPath` context must fall between for the condition to be true. If one of these are omitted, that part of the range will be considered open ended (ie. if no `max` is specified, any input above or equal to `min` will be considered true)
+
+The `inputPath` can be any context that is expressed as a number. Examples of contexts that can be used with the inRange condition are `http://registry.gpii.net/common/environment/auditory.noise` and `http://registry.gpii.net/common/environment/visual.luminosity`, denoting the background noise in dB and light in luminosity.
+
+**Examples:**
+
+Example 1: Any value of visual.luminance above or equal to 700 will be considered true.
+```{
+    "type": "http://registry.gpii.net/conditions/inRange",
+    "min": 700,
+    "inputPath": "http://registry\\.gpii\\.net/common/environment/visual\\.luminance"
+}```
+
+Example 2: Any value of visual.luminance between (incl.) 600 and 1000  will be considered true.
+```{
+    "type": "http://registry.gpii.net/conditions/inRange",
+    "min": 600,
+    "max": 1000,
+    "inputPath": "http://registry\\.gpii\\.net/common/environment/visual\\.luminance"
+}```
+
+Example 3: Any value of auditory.noise below or equal to 100 will be considered true.
+```{
+    "type": "http://registry.gpii.net/conditions/inRange",
+    "max": 100,
+    "inputPath": "http://registry\\.gpii\\.net/common/environment/auditory\\.noise"
+}```
+
+#### http://registry.gpii.net/conditions/timeInRange
+
+This condition is declared using the `http://registry.gpii.net/conditions/timeInRange` type. It is used to express conditions that are related to a timespan. Besides the context, it expects two inputs, both of which much be present. These are `from` and `to` declaring the times which the current time of day must be between (inclusive) for the condition to be evaluated to true. The format of the time should be a string denoting hours (24 hours) and minutes separated by `:`, for example: `"06:30"`, `"19:20"`, `"4:12"`. The condition supports wrapped hours, ie. a `from` time of `"20:00"` and `to` time of `"02:00"`, would evaluate to true between 8 at night and 2 in the morning.
+
+The `inputPath` can be any context that is expressed as unix time (in ms). The one supported by a standard install of the system is `http://registry.gpii.net/common/environment/temporal.time` denoting the current time of day in the user timezone.
```

Comment:
"temporal time" seems an unhelpful pleonasm. A part of the name should give some indication of the real meaning - that is, that it is in the "user's timezone" (whatever that might mean - it might be more helpful to support instead "timezone at the user's current location"). Do we really want users responsible for constantly setting and unsetting their own timezones as they move about? I don't think so.